### PR TITLE
Create base classes for surveyed objects

### PR DESCRIPTION
**Pull request by kaligrafy**

### CODE REVIEW

File: packages/evolution-common/src/services/baseObjects/Surveyable.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+export type SurveyableAttributes = {
+
+    sample: Sample;
+    survey: Survey;
+    sampleBatchNumber?: number;
+
+}
+
+export class Surveyable extends Uuidable {
```
Comment (tahini) [inline packages/evolution-common/src/services/baseObjects/Surveyable.ts:20]: What is this? Who would be surveyable?

File: packages/evolution-common/package.json [hunk 2]
```diff
@@ -24,7 +24,8 @@
     "chaire-lib-common": "^0.2.2",
     "i18next": "^22.4.15",
     "lodash": "^4.17.21",
-    "typescript": "^4.9.4"
+    "typescript": "^4.9.4",
+    "uuid": "^9.0.1"
```
Comment (tahini) [inline packages/evolution-common/package.json:28]: This adds a dependency to yet another uuid package. Please use the same as other packages, or update all of them everywhere (including transition, you might prefer just using the same as elsewhere)

**Review by tahini — CHANGES_REQUESTED** [review]
Please run `yarn format` There seems to be a bit of wrong formatting. Also, that's a lot of types... that don't apply to all surveys. But keep going, it looks like a completely new module! That will be used for the city simulator later on!

File: packages/evolution-common/src/services/baseObjects/BasePerson.ts [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+import { Tripable } from './Tripable';
+import { Uuidable } from './Uuidable';
+import { BasePlace } from './BasePlace';
+import { BaseVisitedPlace } from './BaseVisitedPlace';
+import { BaseTrip } from './BaseTrip';
+import { BaseVehicle } from './BaseVehicle';
+import * as PAttr from './attributeTypes/PersonAttributes';
+import { Vehicleable } from './Vehicleable';
+
+type BasePersonAttributes = {
```
[the remark this answers is not in the document]
Comment (kaligrafy) [reply packages/evolution-common/src/services/baseObjects/BasePerson.ts:19]: I took what is in my thesis, so this is like "official"???!!!

File: packages/evolution-common/src/services/baseObjects/BasePlace.ts [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    _isValid: OptionalValidity;
+
+    geography: GeoJSON.Feature<GeoJSON.Point> | undefined;
+    name?: string;
+    shortname?: string;
+    address?: BaseAddress;
+    osmId?: string; // OpenStreetMap id (can be a node, way or relation). This is the short version: n1234, w2345, r3456
+    landRoleId?: string; // official land role building/location id
+    postalId?: string; // official postal id
+    buildingId?: string; // official building id
```
[the remark this answers is not in the document]
Comment (kaligrafy) [reply packages/evolution-common/src/services/baseObjects/BasePlace.ts:52]: THe official id of the data in the country official database.
Most postal service have specific ids for each address. Same for landrole and the building id, it depends of the country, but some have a specific id for each building.

File: packages/evolution-common/src/services/baseObjects/Surveyable.ts (hunk 1 above)
Comment (kaligrafy) [reply packages/evolution-common/src/services/baseObjects/Surveyable.ts:20]: Added a comment

File: packages/evolution-common/package.json (hunk 2 above)
Comment (kaligrafy) [reply packages/evolution-common/package.json:28]: You mean you want me to update all uuid versions in other packages to 9.0.1? Wouldn't it be better to wait for current surveys to be completed?

File: packages/evolution-common/src/services/baseObjects/Surveyable.ts (hunk 1 above)
Comment (kaligrafy) [reply packages/evolution-common/src/services/baseObjects/Surveyable.ts:26]: BaseINterview is surveyable

**Comment by kaligrafy** [comment]
I added attributes to trip chain (I forgot them)

File: packages/evolution-common/package.json (hunk 2 above)
Comment (kaligrafy) [reply packages/evolution-common/package.json]: I normalized all uuid versions to 8.3.2

**Review by tahini — APPROVED** [review]

**Pull request merged**